component/grant-type/src/main/java/org/wso2/carbon/identity/oauth2/grant/token/exchange/TokenExchangeGrantHandler.java

```diff
@@ -0,0 +1,272 @@
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License
+ */
+
+package org.wso2.carbon.identity.oauth2.grant.token.exchange;
+
+import com.nimbusds.jose.JOSEException;
+import com.nimbusds.jwt.JWTClaimsSet;
+import com.nimbusds.jwt.SignedJWT;
+import org.apache.commons.collections.MapUtils;
+import org.apache.commons.lang.StringUtils;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.wso2.carbon.identity.application.authentication.framework.model.AuthenticatedUser;
+import org.wso2.carbon.identity.application.common.model.ClaimMapping;
+import org.wso2.carbon.identity.application.common.model.IdentityProvider;
+import org.wso2.carbon.identity.core.util.IdentityUtil;
+import org.wso2.carbon.identity.oauth.common.OAuth2ErrorCodes;
+import org.wso2.carbon.identity.oauth.config.OAuthServerConfiguration;
+import org.wso2.carbon.identity.oauth2.IdentityOAuth2Exception;
+import org.wso2.carbon.identity.oauth2.dto.OAuth2AccessTokenRespDTO;
+import org.wso2.carbon.identity.oauth2.grant.token.exchange.utils.TokenExchangeUtils;
+import org.wso2.carbon.identity.oauth2.model.RequestParameter;
+import org.wso2.carbon.identity.oauth2.token.OAuthTokenReqMessageContext;
+import org.wso2.carbon.identity.oauth2.token.handlers.grant.AbstractAuthorizationGrantHandler;
+import org.wso2.carbon.identity.oauth2.util.ClaimsUtil;
+import org.wso2.carbon.identity.oauth2.util.OAuth2Util;
+import org.wso2.carbon.utils.multitenancy.MultitenantConstants;
+
+import java.util.Arrays;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import static org.wso2.carbon.identity.oauth2.grant.token.exchange.utils.TokenExchangeUtils.handleException;
+
+/**
+ * Class to handle Token Exchange grant type
+ */
+public class TokenExchangeGrantHandler extends AbstractAuthorizationGrantHandler {
+
+    private static final Log log = LogFactory.getLog(TokenExchangeGrantHandler.class);
+    private int validityPeriod;
+    private boolean validateIAT = true;
+    private String[] registeredClaimNames = new String[]{"iss", "sub", "aud", "exp", "nbf", "iat", "jti"};
+    private String requested_token_type = TokenExchangeConstants.JWT_TOKEN_TYPE;
+
+    @Override
+    public void init() throws IdentityOAuth2Exception {
+        super.init();
+
+        /*
+          From identity.xml following configs are read.
+
+          <OAuth>
+              <TokenExchangeGrant>
+                  <EnableIATValidation>true</EnableIATValidation>
+                  <IATValidityPeriod>30</IATValidityPeriod>
+              </TokenExchangeGrant>
+          </OAuth>
```

Comment:
IMO, we can remove this comment and add only a comment describing the config. We can move this to readme file

Reply:
Fixed in 8693d5454cb05015e9d9614c56a1d09bdf8a8997